elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/engine/StitchList.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright 2019, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.datastores.aggregation.engine;
+
+import com.yahoo.elide.core.EntityDictionary;
+
+import lombok.AccessLevel;
+import lombok.Data;
+import lombok.Getter;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+
+/**
+ * An auxiliary class for {@link AbstractEntityHydrator} and is responsible for setting relationship values of an entity
+ * instance.
+ * <p>
+ * {@link StitchList} should not be subclassed.
+ */
+public final class StitchList {
+
+    /**
+     * A representation of an TODO item in a {@link StitchList}.
+     */
+    @Data
+    public static class Todo {
+        private final Object entityInstance;
+        private final String relationshipName;
+        private final Object foreignKey;
```

Comment:
Can we change these to `public final` as I don't think we would need to set them?

Reply:
Publisizing non-immutables is dangerous unless there is a compelling reason